assets/src/components/form/select/test/model-select.js

```diff
@@ -15,7 +15,12 @@ const simulatedResponse = [
 describe( 'ModelSelect Snapshot with default options (with required modelName)',
 	() => {
 		it( 'should render and match snapshot', () => {
-			const wrapper = shallow( <ModelSelect modelName={ 'event' } /> );
+			const wrapper = shallow(
+				<ModelSelect
+					modelName={ 'event' }
+					optionsEntityMap={ optionsEntityMap }
```

Comment:
kinda puzzling why this would be needed because the tests were passing in GB master?

Reply:
the `optionsEntityMap` objects are now set on each model, so if you use `EventSelect` then its default `optionsEntityMap` will be passed along to `ModelSelect` for you, but here in this test we are calling  `ModelSelect` directly, so no `optionsEntityMap` is being supplied. 

This first test will actually pass fine without supplying the `optionsEntityMap` because we are also NOT supplying any `modelEntities`.  If you look at the `ModelSelect.getOptions()` method you'll see this:

```js
		if ( isFunction( mapOptionsCallback ) && ! isEmpty( modelEntities ) ) {
			return mapOptionsCallback(
				modelEntities,
				optionsEntityMap,
				mapSelection
			);
		}
		return [];
```

so no `modelEntities` means we don't call the `mapOptionsCallback` (which defaults to `buildOptions` in `assets/src/components/form/select/build-options.js`), but if we DO supply `modelEntities`, then `buildOptions` confirms that we have also supplied a valid  `optionsEntityMap`, and if not displays the following error:

> Warning: A valid optionsEntityMap must be supplied in order to build options for a ModelSelect component

which is what happens now in `assets/src/components/form/select/test/model-select.js` if you supply  `modelEntities` but no `optionsEntityMap`.

With the "child" model selects like `EventSelect`  we don't need to manually supply the `modelEntities` or `optionsEntityMap` because that is all handled automatically for us. They can be overridden, but in those cases you will simply have to supply both. If you want to make things more automatica, then you will likely need to amend `ModelSelect.getOptions()` to retrieve the default `optionsEntityMap` from the specified model if one is not supplied but a set of `modelEntities` is.

Reply:
I understand all that, what I'm getting at though is that the point of the test is just to reflect a snapshot of the shallow rendered `ModelSelect` with _default_ props.  `optionsEntityMap` is not a _required_ prop (unless entities are also provided) and thus the original test was fine as is for the purpose it had.  An example where the test would be useful in its original incarnation is it would fail if `optionsEntityMap` was made required (or the logic in `getOptions` was changed so that it is required).   

A `modelName` was provided in the original incarnation because it's required by virtue of propTypes (enum).  So, based on your explanation, it sounds like you want to change things in `ModelSelect` so that `optionsEntityMap` IS required and thus there'd be validity to you adding it to this test.  In which case, you should also modify `ModelSelect.propTypes` so it has `optionsEntityMap: PropTypes.object.isRequired`

Reply:
`optionsEntityMap` has been removed from the first test which would be evident if you waited for the changes to be pushed and this pull to be reassigned to you before adding jumping in and adding feedback while I'm still working on this. So plz just be patient and you'll get your turn to address things.